src/Nethermind/Nethermind.Merge.Plugin/Synchronization/PeerRefresher.cs

```diff
@@ -17,61 +17,202 @@
     private readonly ISyncPeerPool _syncPeerPool;
     private static readonly TimeSpan _minRefreshDelay = TimeSpan.FromSeconds(10);
     private DateTime _lastRefresh = DateTime.MinValue;
-    private Keccak _lastHash = Keccak.Zero;
+    private (Keccak, Keccak, Keccak) _lastBlockhashes = (Keccak.Zero, Keccak.Zero, Keccak.Zero);
     private readonly ITimer _refreshTimer;
+    private ILogger _logger;
 
-    public PeerRefresher(ISyncPeerPool syncPeerPool, ITimerFactory timerFactory)
+    public PeerRefresher(ISyncPeerPool syncPeerPool, ITimerFactory timerFactory, ILogManager logManager)
     {
         _refreshTimer = timerFactory.CreateTimer(_minRefreshDelay);
         _refreshTimer.Elapsed += TimerOnElapsed;
         _refreshTimer.AutoReset = false;
         _syncPeerPool = syncPeerPool;
+        _logger = logManager.GetClassLogger(GetType());
     }
 
-    public void RefreshPeers(Keccak? blockHash)
+    public void RefreshPeers(Keccak headBlockhash, Keccak headParentBlockhash, Keccak finalizedBlockhash)
     {
-        if (blockHash is not null)
+        _lastBlockhashes = (headBlockhash, headParentBlockhash, finalizedBlockhash);
+        TimeSpan timePassed = DateTime.Now - _lastRefresh;
+        if (timePassed > _minRefreshDelay)
         {
-            _lastHash = blockHash;
-            TimeSpan timePassed = DateTime.Now - _lastRefresh;
-            if (timePassed > _minRefreshDelay)
-            {
-                Refresh(blockHash);
-            }
-            else if (!_refreshTimer.Enabled)
-            {
-                _refreshTimer.Interval = _minRefreshDelay - timePassed;
-                _refreshTimer.Start();
-            }
+            Refresh(headBlockhash, headParentBlockhash, finalizedBlockhash);
+        }
+        else if (!_refreshTimer.Enabled)
+        {
+            _refreshTimer.Interval = _minRefreshDelay - timePassed;
+            _refreshTimer.Start();
         }
     }
 
     private void TimerOnElapsed(object? sender, EventArgs e)
     {
-        Refresh(_lastHash);
+        Refresh(_lastBlockhashes.Item1, _lastBlockhashes.Item2, _lastBlockhashes.Item3);
     }
     
-    private void Refresh(Keccak blockHash)
+    private void Refresh(Keccak headBlockhash, Keccak headParentBlockhash, Keccak finalizedBlockhash)
     {
         _lastRefresh = DateTime.Now;
         foreach (PeerInfo peer in _syncPeerPool.AllPeers)
         {
-            _syncPeerPool.RefreshTotalDifficulty(peer.SyncPeer, blockHash);
+            StartPeerRefreshTask(peer.SyncPeer, headBlockhash, headParentBlockhash, finalizedBlockhash);
         }
     }
 
+    private async Task StartPeerRefreshTask(
+        ISyncPeer syncPeer,
+        Keccak headBlockhash,
+        Keccak headParentBlockhash,
+        Keccak finalizedBlockhash
+    )
+    {
+        CancellationTokenSource delaySource = new();
+        Task delayTask = Task.Delay(RefreshTimeout, delaySource.Token);
```

Comment:
Maybe use: https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtokensource.-ctor?view=net-6.0#system-threading-cancellationtokensource-ctor(system-timespan) instead of separate task? Would make following code easier with just catching cancellation excepiton.